kernel/src/table_changes/mod.rs

```diff
@@ -0,0 +1,196 @@
+//! Provides an API to read the table's change data feed between two versions.
+use std::sync::LazyLock;
+
+use url::Url;
+
+use crate::log_segment::LogSegment;
+use crate::path::AsUrl;
+use crate::schema::{DataType, Schema, StructField, StructType};
+use crate::snapshot::Snapshot;
+use crate::table_features::ColumnMappingMode;
+use crate::{DeltaResult, Engine, Error, Version};
+
+static CDF_FIELDS: LazyLock<[StructField; 3]> = LazyLock::new(|| {
+    [
+        StructField::new("_change_type", DataType::STRING, false),
+        StructField::new("_commit_version", DataType::LONG, false),
+        StructField::new("_commit_timestamp", DataType::TIMESTAMP, false),
```

Comment:
Side note: delta-spark does some hand-wringing about table schemas that already provide columns with these names. I think the solution was to block enabling CDF on such tables, and to block creating columns with those names on tables that are already CDF-enabled? (both are writer issues, not reader, but it's prob worth tracking)

Reply:
Tracked in #524. Thx!